bld/CLMBuildNamelist.pm

```diff
@@ -2125,10 +2125,13 @@ sub setup_logic_roughness_methods {
     $log->fatal_error("$var is incorrect entry for the namelist variable z0param_method; expected Meier2022 or ZengWang2007");
   }
   my $phys = $physv->as_string();
-  if ( $phys eq "clm4_5" || $phys eq "clm5_0" ) {
-    if ( $var eq "Meier2022" ) {
+  if ( $var eq "Meier2022" ) {
+    if ( $phys eq "clm4_5" || $phys eq "clm5_0" ) {
       $log->fatal_error("z0param_method = $var and phys = $phys, but this method has been tested only with clm6_0 and later versions; to use with earlier versions, disable this error, and add Meier2022 parameters to the corresponding params file");
     }
+    if ( &value_is_true($nl_flags->{'use_fates'}) ) {
+      $log->fatal_error("z0param_method = $var and use_fates currently are not compatible.  Please update the z0param_method to ZengWang2007")
```

Comment:
Add a comment that points to the issue to get the details. Probably add the issue number to the error message as well.

Reply:
Should we also disallow this in the FORTRAN code? I vote yes, because it results in bad answers.

Reply:
Oh I see, you mean to update the case code here?

https://github.com/ESCOMP/CTSM/blob/fb89f3359052e4fbba4ff1ccc89e82416a6e544d/src/biogeophys/CanopyFluxesMod.F90#L899-L908

Reply:
It should be something done at initialization rather than runtime. So earlier on.

Reply:
@ekluzek I've added an `endrun` with what I think might be the earliest, pre-existing `use_fates` check during initialization.  Does 43407e1 fit what you were thinking?

Reply:
> Add a comment that points to the issue to get the details. Probably add the issue number to the error message as well.

Addressed per cb361eb

Reply:
This will work. But, a few thoughts. clm_initialize is a pretty high level module. I think it would be good to have it one level lower. Putting specific logic for things like this a level lower is a little more modular.

I looked through the code and have two suggestions. One would be to add this to control_mod.F90 where both use_fates and z0param_method is set. The other would be in init in clmfates_interfaceMod.

Reply:
Thanks for the suggestions @ekluzek.  The fates compatibility section in `control_init` looks good.  I've reverted the previous commit and made the update via ccb55da.